docs/nx-cloud/reference/assignment-rules.md

```diff
@@ -1,39 +1,163 @@
+{% /tab %}
+{% /tabs %}
+
+The first rule will match any task that has a project named `app1`, any targets that begin with `e2e-ci`, and a configuration named `production`. Any tasks that match this rule will only be allowed to run on agents with `linux-medium-js` launch templates. Agents assigned these tasks will also be able to execute up to five tasks in parallel.
+
+The second rule above will match any task that has a `lint` or `build` target. These tasks only run on `linux-large-js` agents and up to ten tasks can be executed in parallel by agents of that type.
 
-You can mix and match any of the criteria in an assignment rule provided that you follow the constraints:
+You can mix and match any of the properties in an assignment rule provided that you follow these constraints:
 
-- At least one of the following properties is defined: `project`, `target`, `configuration`.
-- There is at least one [agent type](/ci/reference/launch-templates) specified in the `runs-on` field.
-- Every changeset in your `distribute-on` field must include at **least one agent** that matches each agent type specified in the `runs-on` field across all assignment rules. For example, if your rules distribute tasks on `linux-small-js`, `linux-medium-js`, and `linux-large-js`, then at least one agent of each type must be available; otherwise, tasks associated with those rules cannot be executed.
+- At least one of the following properties is defined: `projects`, `targets`, `configurations`.
+- There is at least one [agent type](/ci/reference/launch-templates) specified in the `run-on` field. If no parallelism is specified, the parallelism of the executed command will be used instead. If that is not specified, then the parallelism will default to `1`
+- For assignment rules with Nx Agents, every changeset in your `distribute-on` field must include at **least one agent** that matches each agent type specified in the `run-on` field across all assignment rules. For example, if your rules distribute tasks on `linux-small-js`, `linux-medium-js`, and `linux-large-js`, then at least one agent of each type must be available. Otherwise, tasks associated with those rules cannot be executed.
 
 {% callout type="note" title="If you are using self-hosted agents, you must define your own agent types" %}
-You must define your own agent types and attach them to your self-hosted agents using the `NX_AGENT_LAUNCH_TEMPLATE` environment variable. Ensure that for each `runs-on` field in your assignment rules, you have corresponding agents in your agent pool that have the same agent type.
+You must define your own agent types and attach them to your self-hosted agents using the `NX_AGENT_LAUNCH_TEMPLATE` environment variable. Ensure that for each `run-on` field in your assignment rules, you have corresponding agents in your agent pool that have the same agent type.
```

Comment:
This environment variable should also be added to the Cloud environment variables page with a description: https://nx.dev/ci/reference/env-vars